include/fmt/printf.h

```diff
@@ -327,6 +319,37 @@ template <typename T> struct printf_formatter {
   }
 };
 
+// Redirects argument formatting visitor to argument formatters or format arg
+// handler.
+template<typename OutputIt, typename Context, typename FormatSpecs,
+         typename ArgFormatter>
+class printf_arg_formatter_proxy : public internal::function<
```

Comment:
A proxy shouldn't be needed. `ArgFormatter` should be able to deal with both `T` and `handle` case.

Reply:
I agree that this is not really a pretty and clean way. But the problem I tried to solve is, that the printf_arg_formatter did construct a basic_printf_context containing a reference to printf_arg_formatter again. When I try to extend the printf_arg_formatter the basic_printf_context would still be constructed with printf_arg_formatter as template parameter and ignore my child class (and its overloaded **operator()**).

I can also not pass my formatter child class via a template parameter to printf_arg_formatter since I would disable the possibility to specify a default parameter for the forward declaration of printf_arg_formatter (on line 198).

I'm open to implement a complete different approach.

Reply:
`basic_printf_context` should not be parameterized on `ArgFormatter`, it's just an artefact of some earlier design. Please remove it and then you'll be able to address other comments on the PR.

Reply:
When I understand you correctly you mean the ArgFormatter template parameter from basic_printf_context.

That one is still used for the visit_format_arg call in basic_printf_context::format at the end. This was the problem I stumbled over before I decided to make the proxy class because I couldn't come up with another alternative. But I will stick my head into it maybe I've overlooked something or I find a cleaner way.

Would it be better to pass the `ArgFormatter` as template parameter for the format `function` directly?

Reply:
> When I understand you correctly you mean the ArgFormatter template parameter from basic_printf_context.

Yes.

> Would it be better to pass the ArgFormatter as template parameter for the format function directly?

Sure.